Improve documentation for `--reload` and  `--workers` (#1415)
Kludex/uvicorn: docs/deployment.md

When running locally, use `--reload` to turn on auto-reloading.

> Keep in mind, that `--reload` disables `--workers`.

Reviewer: what about something like, "the reload and workers flag are mutually exclusive, use one or the others" ?
the intent is that reload is aimed at development and workers at production, even if I'm not sure why you'd wanna use it but that's another topic.

Author: Yes, makes sense. I can update this.
Background when/why to use both options are in #1413 
(in short: calling the "own" API (via "requests" call, not method calls) blocks the system when not using multiple workers. That's at least my understanding... during development...)

Reviewer: both flags use multiprocess under the hood

Author: I've added a comment to further discuss the issue here: https://github.com/encode/uvicorn/issues/1413#issuecomment-1074948326

Reviewer: not sure if there is a misunderstanding here, this PR is good, just need to rephrase it a bit

Author: Hi @euri10 
sorry for the delay... I pushed to changes. Do you think this is ready?

> The `--reload` and `--workers` arguments are mutually exclusive.

Reviewer 2: The only thing I think should be added on this PR is this line... 🤔

> The `--reload` and `--workers` flag are mutually exclusive. Use one or the other.

Reviewer 2: There's no need for the second sentence.
```suggestion
The `--reload` and `--workers` arguments are mutually exclusive.
```

  --port INTEGER                  Bind socket to this port.  [default: 8000]
  --uds TEXT                      Bind to a UNIX domain socket.
  --fd INTEGER                    Bind to socket from this file descriptor.

>   --reload                        Enable auto-reload. (Disables --workers)

Reviewer: I'm not sure we can have that here since it's automatically generated by click and our `tools.cli_usage` script may remove it
You need anyway to run the script and see, if the script removes it we wont be able and wont add it

Author: Ok, I will check the tool result... thanks for the hint

Author: @euri10 I'm having some trouble starting uvicorn from sources. Any idea how to easily resovle those dependency issues?
```
$ python uvicorn/main.py --help
Traceback (most recent call last):
  File "/workspaces/type-tree/uvicorn/uvicorn/main.py", line 1, in <module>
    import logging
  File "/workspaces/type-tree/uvicorn/uvicorn/logging.py", line 12, in <module>
    class ColourizedFormatter(logging.Formatter):
AttributeError: partially initialized module 'logging' has no attribute 'Formatter' (most likely due to a circular import)
```
I have installed all dependencies from requirements.txt. Any idea?
Thanks!

Reviewer: you just need to run `./scripts/lint` and it will run the tools.cli_usage for you

that error is like wahoo, we should probably rename `logging.py` to avoid namespace conflicts, dont you think @Kludex ?

Reviewer 2: I don't like the name of that module being the same as the stdlib, but I'm fine with it. The error above only happens because of bad usage.

Reviewer 2: If you're strong about it, feel free to change it.

Reviewer: yup never seen that error too, I was just wondering, I think a PR is worth it, I can't count the number of times where I myself got caught bad-naming modules this way, will do it

Author: script executed and changes pushed

Also note that in this case, you should put `uvicorn.run` into `if __name__ == '__main__'` clause in the main module.

> Also in this case, the two options are mutually exclusive, use only one of `reload=True` or `workers=NUM`.

Reviewer 2: ```suggestion
!!! note
`reload` and `workers` parameters are mutually exclusive.
```

Reviewer 2: Please apply this.
